Sources/SwiftProtobuf/_MessageStorage+BinaryDecoding.swift

```diff
@@ -233,6 +360,94 @@ extension _MessageStorage {
         return true
     }
 
+    /// Appends either a single unpacked value or multiple packed values to the array value of the
+    /// given field, depending on the wire format of the corresponding tag.
+    ///
+    /// - Parameters:
+    ///   - reader: The reader from which the values to append should be read.
+    ///   - field: The field being decoded.
+    ///   - tag: The tag that was read from the wire.
+    ///   - unpackedWireFormat: The wire format that should be expected for values of this type if
+    ///     they are unpacked.
+    ///   - decodeElement: A function that takes a `WireFormatReader` covering the slice of packed
+    ///     elements and reads and returns a single element from it.
+    /// - Returns: True if the value was consumed properly, or false if the wire format was not
+    ///   what was expected.
+    private func appendMaybePackedValues<T>(
+        from reader: inout WireFormatReader,
+        to field: FieldLayout,
+        tag: FieldTag,
+        unpackedWireFormat: WireFormat,
+        decodeElement: (inout WireFormatReader) throws -> T
+    ) throws -> Bool {
+        switch tag.wireFormat {
+        case unpackedWireFormat:
```

Comment:
debug assertion that wireformat isn't message/group/bytes/string?

Reply:
Done.